src/windows/models/files_model.py

```diff
@@ -147,7 +147,7 @@ def update_model(self, clear=True):
                 r = get(thumb_address)
                 if r.ok:
                     # Update thumbnail path to real one
-                    thumb_path = r.text
+                    thumb_path = r.content.decode("utf-8", "ignore")
```

Comment:
You want to modify API again...

I think that the p1. is here, (something like `charset=utf-8` can be specified):

https://github.com/OpenShot/openshot-qt/blob/e247a9be079a3628fa046a0f45f2d2825d9aec55/src/classes/thumbnail.py#L156

---------------------

As for the p2. - I see no much difference in this two:
```
response.encoding = 'utf-8'
response.text
```
and the `decode()` code - it surely receives `utf-8`:

https://github.com/OpenShot/openshot-qt/blob/e247a9be079a3628fa046a0f45f2d2825d9aec55/src/classes/thumbnail.py#L184

no need to see the header's charset.

And it can't be safer. It cannot use other technique. I just take example from the https://docs.python.org/3/howto/unicode.html and know the result, but I don't know how the `response.text` works when _...the input string can’t be converted according to the encoding’s rules..._ .